src/Data/StringVariants/NonEmptyText/Internal.hs

```diff
@@ -81,6 +81,13 @@ mkNonEmptyText t
   where
     stripped = T.filter (/= '\NUL') $ T.strip t
 
+mkNonEmptyTextWithTruncate :: forall n. KnownNat n => Text -> Maybe (NonEmptyText n)
+mkNonEmptyTextWithTruncate t
+  | textIsWhitespace stripped = Nothing
+  | otherwise = Just (NonEmptyText $ T.stripEnd $ T.take (fromIntegral $ natVal (Proxy @n)) stripped)
+  where
+    stripped = T.filter (/= '\NUL') $ T.stripStart t
```

Comment:
I feel like this `stripped` function needs to be factored out so we can have consistent use of it - seeing a nearly identical call in the function above makes that seem spurious to me

It *would* be extra work to strip both ends of a `Text` when we're probably going to only take the first N anyway, but we *do* call `stripEnd` above, so I doubt we're saving mcuh time or work here.